plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPropertyPage.java

```diff
@@ -105,69 +110,30 @@ public void run() {
 
   @Override
   public boolean performOk() {
-    // Content can be null if Properties Page is opened and App Engine facets are uninstalled
     if (isValid()) {
-      if (content == null) {
-        return true;
-      } else {
-        return content.savePreferences();
-      }
+      return getActivePanel().savePreferences();
     }
     return false;
   }
 
   @Override
   protected void performDefaults() {
-    content.resetToDefaults();
+    // should we reset all panels?
+    getActivePanel().resetToDefaults();
     super.performDefaults();
   }
 
   @Override
   public void dispose() {
-    if (content != null) {
-      // Content can be null if Properties Page is opened and App Engine facets are uninstalled
-      content.dispose();
-    }
+    blankPreferencesPanel.dispose();
+    standardPreferencesPanel.dispose();
+    flexPreferencesPanel.dispose();
```

Comment:
These panels are `Composite`s, which are anchored to a `PropertyPage`. I think it's not a general practice to explicitly dispose child widgets we create. (For example, we don't dispose `Label`s or `Button`s.) They will be disposed automatically when any of its ancestors is disposed.

OTOH, `databindingSupport` needs to be disposed explicitly.